assets/app/view/game/stock_market.rb

```diff
@@ -338,9 +338,10 @@ def render
           type_text = @game.class::MARKET_TEXT
           colors = @game.class::STOCKMARKET_COLORS
 
-          types_in_market = @game.stock_market.market.flatten.compact.flat_map(&:types)
-          .uniq.to_h { |p| [p, colors[p]] }
+          types_in_market_map = @game.stock_market.market.flatten.compact.flat_map(&:types).uniq
+          types_in_market_map.sort_by! { |e| @game.class::MARKET_TEXT.keys.index(e) }
 
```

Comment:
it also forces games to have a market_text entry for every price type otherwise it will error
should i add || Float::INFINITY to cover for those cases?

Reply:
done, however this will impact games where the MARKET_TEXT is different than the order of prices as appearing the market (46 for example), the legend will be in the wrong order, but this is easily fixed. 
Should I add a flag, or apply it everywhere?